service/src/main/java/io/camunda/service/ElementInstanceServices.java

```diff
@@ -90,6 +109,33 @@ public CompletableFuture<VariableDocumentRecord> setVariables(
     return sendBrokerRequest(brokerRequest);
   }
 
+  private SearchQueryResult<FlowNodeInstanceEntity> toCacheEnrichedResult(
+      final SearchQueryResult<FlowNodeInstanceEntity> result) {
+    final var itemsWithoutName = result.items().stream().filter(u -> !u.hasFlowNodeName()).toList();
+
+    if (itemsWithoutName.isEmpty()) {
+      return result;
+    }
+
+    final var cacheResult =
+        processCache.getCacheItems(
+            itemsWithoutName.stream()
+                .map(FlowNodeInstanceEntity::processDefinitionKey)
+                .collect(Collectors.toSet()));
+
+    return result.withItems(
+        result.items().stream()
+            .map(
+                item ->
+                    item.hasFlowNodeName()
+                        ? item
+                        : item.withFlowNodeName(
```

Comment:
🔧 Do we need to create copies here? Can't we just set the attribute on the entity instance we have?

Reply:
aren't records immutable 🤔, thus I kept here the functional approach which also works much nicer with `.map` (sidenote: I was also thinking about performance here, but I could imagine it does not have a huge impact - but can be wrong)

Reply:
Yeah, they are. I wasn't aware it's a record, actually 😆